packages/presentation/src/useMainDocument.ts

```diff
@@ -157,11 +157,20 @@ export function useMainDocument(props: {
                   document: doc,
                   path: url.pathname,
                 })
-                navigate?.({
-                  id: doc?._id,
-                  type: doc?._type,
-                })
                 mainDocumentIdRef.current = doc?._id
+
+                // We only want to force a navigation to the main document if
+                // the path changed but the document ID did not. An explicit
+                // document navigation should take precedence over displaying
+                // the main document. We determine if an explicit document
+                // navigation has occured by comparing the IDs of the last two
+                // resultant navigation states.
+                if (navigationHistory.at(-1)?.['id'] === navigationHistory.at(-2)?.['id']) {
```

Comment:
Nit: cant you just `.id` here?

Reply:
I wish! But Presentation's tsconfig inherits `noPropertyAccessFromIndexSignature` from `@sanity/pkg-utils` presets, so we'd get a `Property 'id' comes from an index signature, so it must be accessed with ['id'].` error if we tried to do that AFAIK.